src/Kestrel.Core/KestrelServer.cs

```diff
@@ -57,6 +57,11 @@ internal KestrelServer(ITransportFactory transportFactory, ServiceContext servic
             Features = new FeatureCollection();
             _serverAddresses = new ServerAddressesFeature();
             Features.Set(_serverAddresses);
+
+            var initializeAuthority = HttpCharacters.Authority;
```

Comment:
HttpCharacters.Initialize()?

Reply:
I like it. Also, you don't to access all the static fields in HttpCharacters to initialize them. You only need to access one.